src/lib/components/tag/index.tsx

```diff
@@ -0,0 +1,104 @@
+import stylin, { variant } from '@stylin.js/react';
+import { CustomDomComponent, easeInOut, motion } from 'framer-motion';
+import React, { FC, PropsWithChildren } from 'react';
+
+import { Box, Typography } from '../../elements';
+import { TimesSVG } from '../../icons';
+import { colors } from '../../theme/dark/colors';
+import { TagElementProps, TagProps } from './tag.types';
+
+const TagElement = stylin<TagElementProps>('button')(
+  variant({
+    scale: 'tags',
+    property: 'variant',
+  })
+);
+
+const MotionTag = motion(TagElement) as CustomDomComponent<TagProps>;
+
+export const Tag: FC<PropsWithChildren<TagProps>> = ({
+  onClose,
```

Comment:
you can dispense with the use of hasCloseIcon by just having `onClose?: () => void();` and to check whether or not you will have the close icon, you can validate whether the `onClose && <DoingThings />` function exists, if it does then it has an icon, and you avoid using one too many props.